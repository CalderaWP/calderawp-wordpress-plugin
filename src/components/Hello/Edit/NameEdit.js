import React from 'react';
import type {NameEditProps} from "../types";
import {TextControl} from '@wordpress/components'
import type {Node} from 'react';
import {componentClassName} from "../../../control-factory/componentClassName";

export default function (props: NameEditProps): Node {
	const {
		nameControlLabel,
		name,
		onChangeName
	} = props;
	return (
		<TextControl
			className={componentClassName('hello', 'name', 'edit')}
			label={nameControlLabel}
			value={name}
			onChange={onChangeName}
		/>
	);
}


