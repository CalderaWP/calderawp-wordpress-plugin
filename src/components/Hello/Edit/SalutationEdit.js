import React from 'react';
import type {SalutationEditProps} from "../types";
import {TextControl} from '@wordpress/components'
import type {Node} from 'react'
import {componentClassName} from "../../../control-factory/componentClassName";

export default function (props: SalutationEditProps): Node {
	const {
		salutationControlLabel,
		salutation,
		onChangeSalutation
	} = props;
	return (
		<TextControl
			className={componentClassName('hello', 'salutation', 'edit')}
			label={salutationControlLabel}
			value={salutation}
			onChange={onChangeSalutation}
		/>
	);
}


