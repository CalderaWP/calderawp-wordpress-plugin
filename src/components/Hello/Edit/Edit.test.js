import renderer from "react-test-renderer";
import React from "react";
import NameEdit from  './NameEdit';
import SalutationEdit from  './SalutationEdit';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});
describe( 'NameEdit component', () => {
	it( 'matches snapshot', () => {
		expect(
			renderer.create(
				<NameEdit
					name={'Roy'}
					onChangeName={() =>{}}
					nameControlLabel={'Name'}
				/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Has an input that can change', () => {
		let update = '';
		const component = mount(
			<NameEdit
				name={'Roy'}
				onChangeName={(name) =>{
					update=name;
				}}
				nameControlLabel={'Name'}
			/>
		);
		const event = {target: {value: "Mike"}};
		expect(component.children().find('input').length).toBe(1);
		component.children().find('input').simulate('change', event);
		expect(update).toBe('Mike');
	});

});

describe( 'SalutationEdit component', () => {
	it( 'matches snapshot', () => {
		expect(
			renderer.create(
				<SalutationEdit
					salutation={'Hi'}
					salutationControlLabel={'Salutation!'}
					onChangeSalutation={() => {}}
				/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Has an input that can change', () => {
		let update = '';
		const component = mount(
			<SalutationEdit
				salutation={'Hi'}
				salutationControlLabel={'Salutation!'}
				onChangeSalutation={(salutation) => {update = salutation;}}
			/>
		);
		const event = {target: {value: "Hola"}};
		expect(component.children().find('input').length).toBe(1);
		component.children().find('input').simulate('change', event);
		expect(update).toBe('Hola');
	});

});