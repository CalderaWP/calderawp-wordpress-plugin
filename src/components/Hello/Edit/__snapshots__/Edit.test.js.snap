// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameEdit component matches snapshot 1`] = `
<div
  className="components-base-control caldera-name-hello-edit"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-0"
    >
      Name
    </label>
    <input
      aria-describedby={undefined}
      className="components-text-control__input"
      id="inspector-text-control-0"
      onChange={[Function]}
      type="text"
      value="Roy"
    />
  </div>
</div>
`;

exports[`SalutationEdit component matches snapshot 1`] = `
<div
  className="components-base-control caldera-salutation-hello-edit"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-2"
    >
      Salutation!
    </label>
    <input
      aria-describedby={undefined}
      className="components-text-control__input"
      id="inspector-text-control-2"
      onChange={[Function]}
      type="text"
      value="Hi"
    />
  </div>
</div>
`;
